Documentation/auto-dependency-flow/README.md

```diff
@@ -0,0 +1,92 @@
+
+*Repo build*: a particular repository's build that the orchestrator orchestrates. If building the repo involves multiple build legs (e.g. to build native bits on different platforms), all legs together are called *a* repo build.
+
+*Blob feed*: a structured directory of build outputs. AKA "transport feed". Summary of important details for auto dependency flow:
+
+ * There is one blob feed per product build. Repo builds add to the single blob feed.
+ * The orchestrator makes the blob feed available to all repo builds.
+ * The blob feed is on disk or in blob storage virtual directories, but in general it has the same structure either way.
+
+*Upstream/Downstream*: when dicussing repo A, repo B is "upstream" if A has a dependency on product outputs from B. Only repos built by the orchestrator are included. For example, in this doc, external toolsets aren't considered upstream from any repository, nor are any redistributed libraries that we don't build ourselves.
+
+
+# Inserting dependency changes
+
+The dependency change flow is shared by all builds that involve the orchestrator.
+
+TL;DR: the orchestrator derives package version props from the blob feed after each repo build and feeds the data into all subsequent repo builds.
+
+## Vision: product build end-to-end
+This narrative describes an end-to-end Core-Setup (Runtime) product build using CoreFX, CoreCLR, and Standard. More repos will be involved in real product builds.
+
+The orchestrator starts by calculating a repo dependency graph. Edges are determined by comparing each repo's *repo output manifest* and *repo input manifest*. The graph is a DAG, so it's buildable. In this case, we have:
+
+ * Core-Setup
+   * CoreFX
+     * CoreCLR
+     * Standard
+
+The orchestrator sees that CoreCLR and Standard are leaves and kicks off both repo builds. Packages produced [end up on the blob feed](#blob-feed) when they're finished, and CoreCLR and Standard are removed from the graph.
+
+ * Core-Setup
+   * CoreFX
+
+CoreFX is the next leaf, so it will be built next. The orchestrator [creates the package version props based on the blob feed and passes it](#passing-dependency-versions) into a repo build of CoreFX. CoreFX uses the file to determine which package versions to use, and consumes them from the blob feed. The repo build completes and the CoreFX leaf is removed.
+
+ * Core-Setup
+
+The process repeats for the next leaf, Core-Setup: the orchestrator creates the build output props and passes it into a Core-Setup repo build. The Core-Setup build finishes, outputting to the blob feed, and the Core-Setup leaf is removed.
+
+There are no more nodes: the repo builds are complete! The blob feed contains all the product outputs. The orchestrator performs finalization steps (signing, testing, publishing) and then the product build is complete.
```

Comment:
This may be picking on the details, but `signing` at this stage isn't going to work.

Take for example the Core-Setup build today - it needs to produce .zips, .msis, etc. that have assemblies embedded in them.  In order to sign all these different artifacts, we would need to write systems that knows how to take these things apart, sign all the things inside them that need to be signed, and then put them back together again.  Signing is probably best done during the "repo builds" themselves.

Reply:
Same as above, this is something that was discussed earlier on as a way to save time, but I agree I don't see it being feasible (and maybe not even possible, depending on exactly how the layers go together--e.g. signed things put in signed containers would still require multiple round trips).

Also giving removing this a shot.